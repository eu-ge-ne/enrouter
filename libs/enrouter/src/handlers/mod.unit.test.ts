import { describe, test, expect } from "vitest";

import { buildRouteHandlers } from "./mod.js";

import type { Route } from "#route/mod.js";

describe("buildRouteHandlers", () => {
  test("from 1 route", () => {
    const routes: Route = {
      path: "/",
      link: [[], []],
      mod: [],
    };

    expect(buildRouteHandlers({ routes })).toMatchSnapshot();
  });

  test("from 2 routes", () => {
    const routes: Route = {
      path: "/",
      link: [[], []],
      mod: [],
      tree: [
        {
          path: "/abc",
          link: [[], []],
          mod: [],
        },
      ],
    };

    expect(buildRouteHandlers({ routes })).toMatchSnapshot();
  });
});
