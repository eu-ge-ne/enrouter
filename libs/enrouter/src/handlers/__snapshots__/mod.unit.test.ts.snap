// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildRouteHandlers > from 1 route 1`] = `
{
  "modules": [],
  "route": {
    "link": [
      [],
      [],
    ],
    "mod": [],
    "path": "/",
  },
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
}
`;

exports[`buildRouteHandlers > from 2 routes 1`] = `
{
  "modules": [],
  "route": {
    "link": [
      [],
      [],
    ],
    "mod": [],
    "path": "/",
    "tree": [
      {
        "link": [
          [],
          [],
        ],
        "mod": [],
        "path": "/abc",
      },
    ],
  },
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
  "tree": [
    {
      "modules": [],
      "route": {
        "link": [
          [],
          [],
        ],
        "mod": [],
        "path": "/abc",
      },
      "test": {
        "keys": [],
        "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  ],
}
`;
